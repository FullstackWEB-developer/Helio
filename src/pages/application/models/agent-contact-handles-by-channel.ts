import {ChannelTypes} from '@shared/models';

export interface AgentContactHandlesByChannel {
    agent: string;
    channel: ChannelTypes;
    startInterval?: Date;
    endInterval?: Date;
    contactsHandled?: number
}
