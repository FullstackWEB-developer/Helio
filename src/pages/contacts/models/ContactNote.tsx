export interface ContactNote {
    id?: number;
    noteText: string;
    createdOn?: Date;
    createdBy?: string;
}