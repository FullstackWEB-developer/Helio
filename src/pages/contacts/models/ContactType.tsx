export enum ContactType {
    Individual = 'Individual',
    Company = 'Company'
}