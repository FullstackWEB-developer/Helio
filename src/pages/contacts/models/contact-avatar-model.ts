import {Icon} from '@components/svg-icon/icon';
export interface ContactAvatarModel {
    initials?: string;
    className?: string; 
    isCompany?: boolean;
    iconType?: Icon;
    iconClassName?: string;
    iconFillClass?: string;
}