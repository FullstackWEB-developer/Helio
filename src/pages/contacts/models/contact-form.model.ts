import {Option} from "@components/option/option";

export interface ContactFormModel {
    id?: string;
    firstName?: string;
    lastName?: string;
    companyName?: string;
    category: number | Option;
    jobTitle?: string;
    department?: string;
    email?: string;
    workMainPhone?: string;
    workMainExtension?: string;
    workDirectPhone?: string;
    mobile?: string;
    fax?: string;
    website?: string;
    primaryAddressLine?: string;
    primaryApt?: string;
    primaryCity?: string;
    primaryState?: string | Option;
    primaryZipCode?: string;
    shippingAddressLine?: string;
    shippingApt?: string;
    shippingCity?: string;
    shippingState?: string | Option;
    shippingZipCode?: string;
    billingAddressLine?: string;
    billingApt?: string;
    billingCity?: string;
    billingState?: string | Option;
    billingZipCode?: string;
    isStarred?: boolean;
    relatedId?: string;
}