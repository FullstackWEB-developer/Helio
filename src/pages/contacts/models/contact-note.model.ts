export interface ContactNote {
    id?: string;
    noteText: string;
    createdOn: Date;
    createdByName: string;
    contactId: string;
}
export interface AddContactNoteProps {
    contactId: string;
    contactNoteDto: ContactNote
}