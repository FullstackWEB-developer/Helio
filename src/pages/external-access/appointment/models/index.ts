export * from './appointment-type.model';
export * from './appointment-slot.model';
export * from './appointment.model';
export * from './appointment-cancellation.model';
export * from './appointment-cancel-reason.model';
export * from './appointment-slot-request.model';
