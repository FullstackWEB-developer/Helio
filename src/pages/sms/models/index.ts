export * from "./sms-new-message-steps";
export * from "./sms-notification-data";
export * from "./sms-query-type";
