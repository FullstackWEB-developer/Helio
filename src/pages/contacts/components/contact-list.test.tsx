import i18n from '../../../i18nForTests';
import dayjs from 'dayjs';
import duration from 'dayjs/plugin/duration';
import utc from 'dayjs/plugin/utc';
import customParseFormat from 'dayjs/plugin/customParseFormat';
import {unmountComponentAtNode} from 'react-dom';
import {render} from '@testing-library/react';
import TestWrapper from '@shared/test-utils/test-wrapper';
import ContactList from './contact-list';
import Router from "react-router-dom";
import {  Contact, ContactType } from '@shared/models';
jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useParams: jest.fn(),
   }));
describe("ContactList tests", () => {
    let container: HTMLDivElement | null;
    const mockState = {
        emailState: {
            unreadEmails: 0,
            messageSummaries: []
        },
        appUserState: {
            liveAgentStatuses: [],
            appUserDetails: {
                id: ""
            },
            auth: {
                name: "Test"
            }
        },
        lookupsState: {

        },
        ticketState: {
            lookupValues: [{
                label: "Test",
                parentValue: "Test"
            },{
                label: "Test",
                parentValue: "Test"
            }]
        },
        appState: {
            smsTemplates: [],
            emailTemplates: []
        },
        ccpState: {
            chatCounter: 1,
            voiceCounter: 2
        }
    };

    beforeEach(async () => {
        await i18n.init();
        dayjs.extend(duration);
        dayjs.extend(utc);
        dayjs.extend(customParseFormat);
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        if (container) {
            unmountComponentAtNode(container);
            container.remove();
            container = null;
        }
    });

    it("renders ContactList return correctly", async () => {
        let contact = {
            category: 1,
            companyName: "",
            firstName: "",
            id: "",
            lastName: "",
            mobilePhone: "",
            name: "",
            type: ContactType.Individual,
            workMainPhone: "",
        } as Contact;
        jest.spyOn(Router, 'useParams').mockReturnValue({ id: '1234' })
        const {asFragment} = render(<TestWrapper mockState={mockState}>
            <ContactList contacts={[contact]} onContactSelect={() => {}} handleAddNewContactClick={() => {}} fetchMore={() => {}} isFetchingNextPage={true} isFetching={true} searchValue={""} searchHandler={() => {}}/>
        </TestWrapper>);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders ContactList empty return correctly", async () => {
        jest.spyOn(Router, 'useParams').mockReturnValue({ id: '1234' })
        const {asFragment} = render(<TestWrapper mockState={mockState}>
            <ContactList contacts={[]} onContactSelect={() => {}} handleAddNewContactClick={() => {}} fetchMore={() => {}} isFetchingNextPage={true} isFetching={true} searchValue={""} searchHandler={() => {}}/>
        </TestWrapper>);
        expect(asFragment()).toMatchSnapshot();
    });
})
