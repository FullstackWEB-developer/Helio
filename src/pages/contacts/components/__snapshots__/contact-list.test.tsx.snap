// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactList tests renders ContactList empty return correctly 1`] = `
<DocumentFragment>
  <div
    class="w-72 overflow-x-hidden relative flex flex-col contact-list-section"
  >
    <div
      class="w-full"
    >
      <div
        class="flex items-center pl-4 py-3 cursor-pointer align-middle border-b"
      >
        <div
          class=""
        >
          <svg
            class="icon-large pl-1 cursor-pointer "
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="24"
              width="24"
            />
            <path
              class="active-item-icon"
              d="M15,12a4,4,0,1,0-4-4A4,4,0,0,0,15,12ZM6,10V7H4v3H1v2H4v3H6V12H9V10Zm9,4c-2.67,0-8,1.34-8,4v2H23V18C23,15.34,17.67,14,15,14Z"
            />
          </svg>
        </div>
        <span
          class="body2 pl-4 contact-accent-color"
        >
          contacts.contact-list.add
        </span>
      </div>
    </div>
    <div
      class="relative h-10 w-full search-input-field flex items-center pl-2 py-6"
    >
      <div
        class="absolute pl-4"
      >
        <div
          class=""
        >
          <svg
            class="cursor-pointer icon-small "
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="16"
              width="16"
            />
            <path
              class="search-icon-fill"
              d="M12.7,11.23a6.777,6.777,0,0,0,1.4-4.174A7.02,7.02,0,0,0,7.1,0,7.105,7.105,0,0,0,0,7.056a7.105,7.105,0,0,0,7.1,7.056,6.667,6.667,0,0,0,4.2-1.391l3,2.981a.971.971,0,0,0,1.4,0,.957.957,0,0,0,0-1.391Zm-5.6.8A5.022,5.022,0,0,1,2,7.056a5.1,5.1,0,0,1,10.2,0A5.022,5.022,0,0,1,7.1,12.025Z"
            />
          </svg>
        </div>
      </div>
      <input
        autocomplete="off"
        class="py-2.5 h-full w-full search-input-field body2 contact-search-input-field pl-12 border-b"
        placeholder="contacts.contact-list.search"
        type="text"
        value=""
      />
    </div>
    <div
      class="relative max-w-full overflow-y-auto overflow-x-hidden h-full"
    >
      <div
        class="subtitle3-small w-full text-center mt-5"
      >
        contacts.contact-list.no_results
      </div>
      <div
        class="visible"
      >
        <div
          class="flex flex-col space-y-4 items-center h-full w-full justify-center undefined"
        >
          <div>
            <div
              class=""
            >
              <svg
                class="spinner icon-large-40 "
                viewBox="0 0 40 40"
              >
                <circle
                  class="path"
                  cx="20"
                  cy="20"
                  fill="none"
                  r="16"
                  stroke-width="4"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ContactList tests renders ContactList return correctly 1`] = `
<DocumentFragment>
  <div
    class="w-72 overflow-x-hidden relative flex flex-col contact-list-section"
  >
    <div
      class="w-full"
    >
      <div
        class="flex items-center pl-4 py-3 cursor-pointer align-middle border-b"
      >
        <div
          class=""
        >
          <svg
            class="icon-large pl-1 cursor-pointer "
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="24"
              width="24"
            />
            <path
              class="active-item-icon"
              d="M15,12a4,4,0,1,0-4-4A4,4,0,0,0,15,12ZM6,10V7H4v3H1v2H4v3H6V12H9V10Zm9,4c-2.67,0-8,1.34-8,4v2H23V18C23,15.34,17.67,14,15,14Z"
            />
          </svg>
        </div>
        <span
          class="body2 pl-4 contact-accent-color"
        >
          contacts.contact-list.add
        </span>
      </div>
    </div>
    <div
      class="relative h-10 w-full search-input-field flex items-center pl-2 py-6"
    >
      <div
        class="absolute pl-4"
      >
        <div
          class=""
        >
          <svg
            class="cursor-pointer icon-small "
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="16"
              width="16"
            />
            <path
              class="search-icon-fill"
              d="M12.7,11.23a6.777,6.777,0,0,0,1.4-4.174A7.02,7.02,0,0,0,7.1,0,7.105,7.105,0,0,0,0,7.056a7.105,7.105,0,0,0,7.1,7.056,6.667,6.667,0,0,0,4.2-1.391l3,2.981a.971.971,0,0,0,1.4,0,.957.957,0,0,0,0-1.391Zm-5.6.8A5.022,5.022,0,0,1,2,7.056a5.1,5.1,0,0,1,10.2,0A5.022,5.022,0,0,1,7.1,12.025Z"
            />
          </svg>
        </div>
      </div>
      <input
        autocomplete="off"
        class="py-2.5 h-full w-full search-input-field body2 contact-search-input-field pl-12 border-b"
        placeholder="contacts.contact-list.search"
        type="text"
        value=""
      />
    </div>
    <div
      class="relative max-w-full overflow-y-auto overflow-x-hidden h-full"
    >
      <div
        class="w-full h-6 company-item-letter pl-6 body2"
      />
      <div
        class="max-h-10 max-w-full truncate body2 py-2.5 relative border-b cursor-pointer pl-6"
        data-testid=""
      >
        <div
          class="pr-10 truncate "
        >
           
        </div>
      </div>
      <div
        class="visible"
      >
        <div
          class="flex flex-col space-y-4 items-center h-full w-full justify-center undefined"
        >
          <div>
            <div
              class=""
            >
              <svg
                class="spinner icon-large-40 "
                viewBox="0 0 40 40"
              >
                <circle
                  class="path"
                  cx="20"
                  cy="20"
                  fill="none"
                  r="16"
                  stroke-width="4"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
